python/strings_udf/strings_udf/tests/utils.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+import numba
+from numba.core.typing.templates import AbstractTemplate
+from numba.cuda.cudadecl import registry as cuda_decl_registry
+from numba.cuda.cudaimpl import lower as cuda_lower
+
+from strings_udf._typing import StringView, string_view, udf_string
+from strings_udf.lowering import cast_string_view_to_udf_string
+
+
+def sv_to_udf_str(sv):
+    pass
```

Comment:
It looks like this function is already defined here so I'm not sure why it's also needed in the main codebase. Maybe that can just be removed?

Reply:
This is a great question. It happens as follows. All strings in `strings_udf` start life as a `string_view`, because the input column isn't being mutated. `column_to_string_view_array` handles this. It stays that way all the way until it hits the users UDF. Imagine they have something like this:

```python
def f(some_string: string_view):
    return some_string.startswith('a')
```

Following the typing in this case, `some_string` is known to be of type `string_view`. When `startswith` is called, numba searches for a declaration of `StringView.startswith(StringView)` and finds the one that returns a `bool`. The lowering to actually do the work is inserted there and everything is fine. 

Now, suppose the user has something like

```python
def f(some_string: string_view):
    new_string = some_string.upper()
    return new_string.startswith('a')
```

Again `some_string` starts life as an instance of `StringView`, but that's where the similarities end. In this case, numba needs to figure out what type `new_string` is. Since functions that return strings need to be built around a `udf_string`, that's what `upper()` does. Numba will find the overload of `StringView.upper() -> udf_string` and determine that the type of `new_string` is now `udf_string`. 

This is where we hit the problem because there _is no method_ `UDFString.startswith`. In the c++ layer there isn't even a `startswith` method associated with `udf_string` objects because they're considered to be a read-only operation, just like there's no `upper()` method of `cudf::string_view`. 

What this PR is trying to do is turn one into the other when necessary. When a `string_view` tries to call a `udf_string` method, it needs to be cast, and the same way the other way around.